scopes/lanes/lanes/lanes.main.runtime.ts

```diff
@@ -187,6 +197,110 @@ export class LanesMain {
+    }
+
+    const laneId: LaneId = (laneName && LaneId.from(laneName)) || LaneId.from(currentLaneName as string);
+    const scope: Scope = this.workspace.scope.legacyScope;
+    const lane: Lane | null | undefined = await scope.loadLane(laneId);
+
+    if (!lane?.readmeComponent) {
+      return {
+        result: false,
+        message: `there is no readme component added to the lane ${laneName || currentLaneName}`,
+      };
+    }
+
+    const readmeComponentId = await this.workspace.resolveComponentId(lane.readmeComponent.id);
+    const existingLaneConfig =
+      (await this.workspace.getSpecificComponentConfig(readmeComponentId, LanesAspect.id)) || {};
+
+    if (existingLaneConfig !== '-') {
+      delete existingLaneConfig[lane.name];
+      // this.workspace.bitMap.removeComponentConfig(readmeComponentId, LanesAspect.id, false);
+      await this.workspace.removeSpecificComponentConfig(readmeComponentId, LanesAspect.id, false);
+      await this.workspace.addSpecificComponentConfig(readmeComponentId, LanesAspect.id, existingLaneConfig);
+    } else {
+      // this should not happen but is it still possible to set the config as "-"
+      await this.workspace.removeSpecificComponentConfig(readmeComponentId, LanesAspect.id, false);
+    }
+    lane.setReadmeComponent(undefined);
+    await scope.lanes.saveLane(lane);
+    await this.workspace.bitMap.write();
+
+    return { result: true };
+  }
+
+  async addLaneReadme(readmeComponentIdStr: string, laneName?: string): Promise<{ result: boolean; message?: string }> {
+    if (!this.workspace) {
+      throw new BitError(`unable to track a lane readme component outside of Bit workspace`);
+    }
+    const readmeComponentId = await this.workspace.resolveComponentId(readmeComponentIdStr);
+
+    const currentLaneName = this.getCurrentLane();
+
+    const readmeComponentBitId = readmeComponentId._legacy;
+
+    const laneId: LaneId = (laneName && LaneId.from(laneName)) || LaneId.from(currentLaneName as string);
+    const scope: Scope = this.workspace.scope.legacyScope;
+    const lane: Lane | null | undefined = await scope.loadLane(laneId);
+
+    if (!lane) {
+      return { result: false, message: `cannot find lane ${laneName}` };
+    }
+
+    lane.setReadmeComponent(readmeComponentBitId);
+    await scope.lanes.saveLane(lane);
+
+    // const existingLaneConfig = this.workspace.bitMap.getBitmapEntry(readmeComponentId)?.config?.[LanesAspect.id] || {};
+    const existingLaneConfig =
+      (await this.workspace.getSpecificComponentConfig(readmeComponentId, LanesAspect.id)) || {};
+    if (existingLaneConfig !== '-') {
+      await this.workspace.addSpecificComponentConfig(readmeComponentId, LanesAspect.id, {
+        ...existingLaneConfig,
+        [lane.name]: { readme: true },
+      });
+    } else {
+      // this should not happen but is it still possible to set the config as "-"
+      await this.workspace.addSpecificComponentConfig(readmeComponentId, LanesAspect.id, {
```

Comment:
what is the lane id here?
is it the local land id? the remote one? they might be different between machines if I remember correctly

Reply:
the problem is that I'm exporting the lane and the component, then in another workspace, someone will bring both, but use another name of the lane, so you won't identify it. 
it's connected to the fact that the component is aware of a specific lane.